flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandler.java

```diff
@@ -306,6 +306,7 @@ private void decodeBufferOrEvent(RemoteInputChannel inputChannel, NettyMessage.B
 				Buffer buffer = inputChannel.requestBuffer();
 				if (buffer != null) {
 					nettyBuffer.readBytes(buffer.asByteBuf(), receivedSize);
+					buffer.setCompressed(bufferOrEvent.isCompressed);
```

Comment:
we can avoid this via https://github.com/apache/flink/pull/10375/commits/2f538ab20262fbe9981d4851b93e9bee04fc6ad0#r353537910